src/components/UI/navbar/index.js

```diff
@@ -1,50 +1,231 @@
-              style={{ color: '#87D870' }}
+    <div className={styles.wrapper}>
+      <>
+        <nav className={styles.navBar}>
+          <div
+            className={styles.hamburger}
+            ref={navbarRef}
+            onClick={() => setToggle(!toggle)}
+            onKeyDown={() => setToggle(!toggle)}
+          >
+            <span className={styles.bar} />
+            <span className={styles.bar} />
+            <span className={styles.bar} />
+          </div>
+          <div
+            className={
+              mountedComponent
+                ? `${styles.navBarLogin}`
+                : `${styles.navBarLogin} d-none`
+            }
+          >
+            <Link href={LOGIN_URL}>
+              <a
+                className={
+                  isLoggedIn
+                    ? `${styles.btnLogin} d-none`
+                    : `${styles.btnLogin}`
+                }
+              >
+                <button type="button" className={styles.btnLoginText}>
+                  Sign In
+                  <img
+                    className={styles.githubLogo}
+                    src={GITHUB_LOGO}
+                    alt="GitHub Icon"
+                    height="15px"
+                    width="15px"
+                  />
+                </button>
+              </a>
+            </Link>
+            <div
+              className={
+                isLoggedIn
+                  ? `${styles.userGreet}`
+                  : `${styles.userGreet} d-none`
+              }
+            >
+              <Link href={USER_PROFILE_URL}>
+                <a>
+                  <div className={styles.userGreetMsg}>
+                    {isLoggedIn
+                      ? `Hello, ${userData.firstName}!`
+                      : `Hello, User!`}
+                  </div>
+                  <img
+                    className={styles.userProfilePic}
+                    src={
+                      isLoggedIn
+                        ? `https://raw.githubusercontent.com/Real-Dev-Squad/website-static/main/members/${userData.userName}/img.png`
+                        : `${DEFAULT_AVATAR}`
+                    }
+                    alt="Profile pic"
+                  />
+                </a>
+              </Link>
+            </div>
+          </div>
+          <ul
+            className={
+              toggle
+                ? `${styles.navBarMenu} ${styles.active}`
+                : styles.navBarMenu
+            }
+          >
+            <li className={styles.navBarLogoLi}>
+              <a href={PATHS.HOME}>
+                <Image
+                  width="50px"
+                  height="50px"
+                  src="/images/Real-Dev-Squad@1x.png"
+                  alt="real-dev-squad"
+                />
+              </a>
+            </li>
+            <li>
+              <Link href={PATHS.WELCOME}>Welcome</Link>
+            </li>
+            <li>
+              <Link href={PATHS.EVENTS}>Events</Link>
```

Comment:
@rohan09-raj do you have some suggestion here? Familiar scenario?

Reply:
Yeah, I have asked him to use an Array of objects here as the repetition of code is there.
@Kshashank99 Can you please do the needful ??